smartapps/initialstate-events/initial-state-event-streamer.src/initial-state-event-streamer.groovy

```diff
@@ -331,58 +330,57 @@ def genericHandler(evt) {
 // after a specified amount of time to reduce the load on ST servers
 def flushBuffer() {
 	log.trace "About to flush the buffer on schedule"
-	if (atomicState.eventBuffer != null && atomicState.eventBuffer.size() > 0) {
+	if (state.eventBuffer != null && state.eventBuffer.size() > 0) {
 		tryShipEvents()
 	}
 }
 
 def eventHandler(name, value) {
-	log.debug atomicState.eventBuffer
-
 	def eventBuffer = atomicState.eventBuffer
-	def epoch = now() / 1000
-	
-	// if for some reason this code block is being run
-	// but the SmartApp wasn't propery setup during install
-	// we need to set initialize the eventBuffer.
-	if (!atomicState.eventBuffer) {
-		atomicState.eventBuffer = []
+	if (!eventBuffer) {
+		// Initialize the eventBuffer since the SmartApp wasn't setup properly
+		eventBuffer = []
 	}
 	eventBuffer << [key: "$name", value: "$value", epoch: "$epoch"]
-	
-	log.debug eventBuffer
-
 	atomicState.eventBuffer = eventBuffer
 
-	if (eventBuffer.size() >= 10) {
+	log.debug eventBuffer
+
+	// Try to post to Initial State every 10 events, doing mod in case there are failures and we didn't flush
+	if (eventBuffer.size() % 10 == 0) {
 		tryShipEvents()
 	}
 }
 
 // a helper function for shipping the atomicState.eventBuffer to Initial State
 def tryShipEvents() {
-
 	// can't ship events if there is no grokerSubdomain
-	if (atomicState.grokerSubdomain == null || atomicState.grokerSubdomain == "") {
+	if (state.grokerSubdomain == null || state.grokerSubdomain == "") {
```

Comment:
if the `grokerSubdomain` is going to be used here in the evaluation, it needs to be set at some point in the app, this commit is only setting the `atomicState.grokerSubdomain` so this conditional will always evaluate as true and no events will ever be shipped because the url will not be correct.